test/auth_test.py

```diff
@@ -209,52 +213,73 @@ def test_transfered_ongoing_non_continues_ops(cfg):
 
 
 @pytest.mark.benchmark
-def test_run_operation_benchmark(cfg):
-    # Run 1000000 operations with 4 concurrent threads.
+@pytest.mark.parametrize("workers", [1, 2, 4, 8])
+def test_run_operation_benchmark(cfg, workers):
+    # Simulate tranferr of a big image using nbdcopy as a client. We split the
+    # image to 128 MiB segments. Every wroker send one segment and get the next
+    # segment to transfer.
+
     ticket = Ticket(testutil.create_ticket(ops=["read"]), cfg)
-    operations = 10**6
-    workers = 4
-    chunk = 10**9
-    step = chunk * workers // operations
 
-    def worker(offset, size):
-        while offset < size:
-            ticket.run(Operation(offset, step))
-            offset += step
+    image_size = 100 * 1024**3
+    segment_size = 128 * 1024**2
+    io_size = 2 * 1024**2
+
+    segment_count = (image_size + segment_size - 1) / segment_size
```

Comment:
I'm vaguely starting to recall the Python 3 porting work :)
Wouldn't that result a "float"? Thus you need to use "//".
Or even better use:
math.ceil(image_size/segment_size)

Reply:
Yes, should be fixed.